### DISCUSSION TITLE
fileClass select type for Tags

### DISCUSSION THREAD

**Discussion by ChristinWhite** [mdelobelle/metadatamenu#354] [Tips and Tricks]
Is there a good way to create a select type in a fileClass for dynamically grabbing tags? I tried using the following in the DataView function but it doesn't return any results. Not being a DataView power user by any stretch I'm likely doing it wrong.

```
dv.list(new Set(dv.pages().file.tags))
```

Ideally I'd love to be able to do one version that grabs all tags and another that grabs all subtags under `#Class`.

**Comment by johnkferguson** [comment]
Hey,

I had issues with this too, but was able to get it to work.

Here's what I learned. 

You're not supposed to use `dv.list` when using a dataview function within metadata menu. Just using `dv.pages` is enough.

To get all the distinct tags in your vault and display for use with metadata menu, you can say, `dv.pages().file.tags.distinct()`

However, this will include the hashtag in the tag name. If you add a tag with a hashtag in it to the `tags' field in your frontmatter, it won't show up as a valid tag when looking for that tag. So you will need to remove the `#`.

This can be done with:
`dv.pages().file.tags.distinct().map(str => str.replace(/#/g, ''))`

When you click on the metadata menu icon next to your file name and select the tags field (after setting this up), you should be able to see a list of all your tags and can then select them.
